fgpyo/fasta/builder.py

```diff
@@ -0,0 +1,231 @@
+"""
+Classes for generating fasta files and records for testing
+----------------------------------------------------------
+This module contains utility classes for creating fasta files, indexed fasta files (.fai), and
+sequence dictionaries (.dict).
+
+Examples of creating sets of contigs for writing to fasta
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Writing a FASTA with two contigs each with 100 bases.
+.. code-block:: python
+    >>> from fgpyo.fasta.builder import FastaBuilder
+    >>> builder = FastaBuilder()
+    >>> builder.add("chr10").add("AAAAAAAAAA", 10)
+    >>> builder.add("chr11").add("GGGGGGGGGG", 10)
+    >>> builder.to_file(path = pathlib.Path("test.fasta"))
+Writing a FASTA with one contig with 100 A's and 50 T's
+    >>> from fgpyo.fasta.builder import FastaBuilder
+    >>> builder = FastaBuilder()
+    >>> builder.add("chr10").add("AAAAAAAAAA", 10).add("TTTTTTTTTT", 5)
+    >>> builder.to_file(path = pathlib.Path("test.fasta"))
+Add bases to existing contig
+    >>> from fgpyo.fasta.builder import FastaBuilder
+    >>> builder = FastaBuilder()
+    >>> contig_one = builder.add("chr10").add("AAAAAAAAAA", 1)
+    >>> contig_one.add("NNN", 1)
+    >>> contig_one.bases
+    'AAAAAAAAAANNN'
+"""
+import textwrap
+from pathlib import Path
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+
+""" Stubs for pysam imports """
+if TYPE_CHECKING:
+
+    def dict(*args: Any) -> None:
+        pass
+
+
+else:
+    from pysam import dict
```

Comment:
Lots of extra blank lines here!

Reply:
Also given the `dict()` builtin, I wouldn't call it this.  Perhaps:

```suggestion
if TYPE_CHECKING:
    def samtools_dict(*args: Any) -> None:
        pass
else:
    from pysam import dict as samtools_dict
```

Reply:
I think the blank lines are required for `black`